packages/kbn-management/settings/components/form/types.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  FieldRowKibanaDependencies,
+  FieldRowServices,
+} from '@kbn/management-settings-components-field-row';
+import { SettingType, UnsavedFieldChange } from '@kbn/management-settings-types';
+import { SettingsStart } from '@kbn/core-ui-settings-browser';
+import { I18nStart } from '@kbn/core-i18n-browser';
+import { ThemeServiceStart } from '@kbn/core-theme-browser';
+
+/**
+ * Contextual services used by a {@link Form} component.
+ */
+export interface Services {
+  saveChanges: (changes: Record<string, UnsavedFieldChange<SettingType>>) => void;
+  showError: (message: string) => void;
+  showReloadPagePrompt: () => void;
+}
+
+/**
+ * Contextual services used by a {@link Form} component and its dependents.
+ */
+export type FormServices = FieldRowServices & Services;
+
+/**
+ * An interface containing a collection of Kibana plugins and services required to
+ * render a {@link Form} component.
+ */
+export interface KibanaDependencies {
```

Comment:
nit: while I know `FieldRowServices` includes the `toast` and other interfaces you use in `services.tsx`, they need to be explicitly stated here, since this component uses them.

also, no need to export this, as `FormKibanaDependencies` encapsulates the total dependency interface.

Reply:
Ah I didn't know this, thanks for letting me know! Addressed in https://github.com/elastic/kibana/pull/166460/commits/75c941ee5d83c00ea6d34a2cdae01a0a4ee938a3